src/InvestJob.js

```diff
@@ -0,0 +1,119 @@
+import crypto from 'crypto';
+import localforage from 'localforage';
+
+import { getLogger } from './logger';
+
+const logger = getLogger(__filename.split('/').slice(-1)[0]);
+
+const KEYS_ARRAY = 'sortedHashArray';
+
+/**
+ * Create an object to hold properties associated with an Invest Job.
+ *
+ */
+export default class InvestJob {
+  /* If none exists, init an empty array for the sorted workspace hashes */
+  static async initDB() {
+    const keys = await localforage.getItem(KEYS_ARRAY);
+    if (!keys) {
+      localforage.setItem(KEYS_ARRAY, []);
+    }
+  }
+
+  /* Return an array of job metadata objects, ordered by most recently saved */
+  static async getJobStore() {
+    let jobArray = [];
+    const sortedKeys = await localforage.getItem(KEYS_ARRAY);
+    if (sortedKeys) {
+      jobArray = await Promise.all(sortedKeys.map(
+        (key) => localforage.getItem(key)
+      ));
+    }
+    return jobArray;
+  }
+
+  static async clearStore() {
+    await localforage.clear();
+  }
+
+  /**
+   * @param {object} obj - the metadata property
+   * @param {string} obj.modelRunName - name to be passed to `invest run`
+   * @param {string} obj.modelHumanName - colloquial name of the invest model
+   * @param {object} obj.argsValues - an invest "args dict" with initial values
+   * @param {object} obj.workspace - defines the invest workspace and suffix
+   * @param {string} obj.workspace.directory - path to invest model workspace
+   * @param {string} obj.workspace.suffix - invest model results suffix
+   * @param {string} obj.logfile - path to an existing invest logfile
+   * @param {'running'|'success'|'error'} obj.status - status of the invest process
```

Comment:
Having this info is cool! Just out of curiosity what are the cases where I'd load a model back up and see `running`?

Reply:
The status is updated when the run button is clicked (to 'running') and then again when invest exits (to either 'sucess' or 'error'). And the `save()` method should be called at both of those times as well. So if you were to switch back to the Home Tab while the model was running, the card should display with a spinner.

But, I just tested this and I think I have found a related bug that I introduced in this refactor. So I'd better fix that!

Reply:
Tangentially related to this, I removed the Spinner - I find them slightly frustrating to have on the screen - and replaced with text of the status itself if it is 'running' or 'error'.